### DISCUSSION TITLE
Some artefacts in coupled phase-field-mechanics task

### DISCUSSION THREAD

**Discussion by RossMech** [idaholab/moose#17652] [Q&A Modules: Phase field] [answered]
Hello everyone,

I have some numerical or model artefact in coupled phase-field and mechanical model. The value of order parameter is exceeding [0..1] range locally and seems to be some kind of the non-physical oscillation.

[
![Screenshot_1](https://user-images.githubusercontent.com/62218067/115377849-4a7e6900-a1d0-11eb-8d24-edd6d9e85a30.png)
](url)

The model is multi-phase model with penalty stabilization.

What have I done in troubleshooting

1. The automatic scaling of the residuals
2. Mesh size
3.  The part of conserved and non-conserved variables in interfacial energy
4. Change the form of local free energy functions (different polynomials and cosh)
5. Time step reduction
6. Penalty size

Time step reduction seems to improve the situation, but the step size is then quite small.

Would like to discuss the ideas about why it happens and what troubleshooting strategy can be used

Best regards,
Rostyslav Nizinkovskyi.

**Comment by hugary1995** [comment, marked as answer]
Can you try adding bound constraints on the order parameter? [https://mooseframework.inl.gov/source/auxkernels/ConstantBoundsAux.html](https://mooseframework.inl.gov/source/auxkernels/ConstantBoundsAux.html) 

After adding the bounds, i.e. a lower bound of 0 and an upper bound of 1, make sure you use one of the two VI solvers, vinewtonssls or vinewtonrsls.

**Reply by RossMech** [reply]
Hello, thank you for the response and suggestion. I would try to avoid it, because the task with lagrange multiplier would be very stiff and the time step would decrease drastically as well. But if nothing seems to work, I would try to use this AuxKernel then.

**Reply by hugary1995** [reply]
> I would try to avoid it, because the task with lagrange multiplier would be very stiff and the time step would decrease drastically as well.

I'm not convinced that's what will happen here. But anyways feel free to try other methods.

I don't think the oscillations can be completely "fixed" if you keep using the penalty approach.

**Comment by laagesen** [comment]
You mentioned this is a coupled phase-field/mechanics simulation...I would try removing the mechanics part of the problem and see if the issue is still there. The shape of your particle is not circular, if you remove the mechanics part (and you are using isotropic interfacial energy) then you should eventually reach the energy-minimizing circular shape, and depending on the formulation I would not expect order parameter values greater than 1. I would make sure this happens first. Either way it will help diagnose where the problem lies.
